plugins/ngrok/provisioner.go

```diff
@@ -24,7 +27,38 @@ const (
 type fileProvisioner struct {
 }
 
-func newNgrokProvisioner() sdk.Provisioner {
+func ngrokProvisioner() sdk.Provisioner {
+	cmd := exec.Command("ngrok", "--version")
+	ngrokVersion, err := cmd.Output()
+	if err != nil {
+		return fileProvisioner{}
+	}
+
+	// Example: "ngrok version 3.1.1\n" to "3.1.1\n"
+	currentVersion := strings.TrimPrefix(string(ngrokVersion), "ngrok version ")
+
+	// Example: "3.1.1\n" to "3.1.1"
+	currentVersion = strings.Trim(currentVersion, "\n")
+
+	// Example: "3.1.1" to "v3.1.1" as that's the format
+	// the package semver expects
+	currentVersion = "v" + currentVersion
+
+	// NGROK_API_KEY is supported only from ngrok 3.2.1
+	// NGROK_AUTH_TOKEN was already supported
+	requiredVersion := "v3.2.1"
+
+	// If the current ngrok CLI version is 3.2.1 or higher,
+	// use environment variables to provision the Shell Plugin credentials
+	//
+	// semver.Compare resulting in 0 means 3.2.1 is in use
+	// semver.Compare resulting in +1 means >3.2.1 is in use
+	if semver.Compare(currentVersion, requiredVersion) == 0 || semver.Compare(currentVersion, requiredVersion) == +1 {
+		return provision.EnvVars(defaultEnvVarMapping)
```

Comment:
@arunsathiya  another thing I'd like to see tackled in this PR, if you have enough time before a security/Sr. review is performed on this PR, is to :
1. Create a wrapper env var provisioner for ngrok. In its provision function it will check the version (as you already do here) but it will return an error if the version is lower than 3.2.1.
2. Add this wrapper env var provisioner as the `DefaultProvisioner`, while using this function `ngrokProvisioner()` in the credential Usage definition of the `ngrok` executable definition. 

What this means is that: 
- ngrok and only ngrok will use either the file provisioner or the env var provisioner
- any other tool that will use ngrok's credential type in the future will only use the env var provisioner. Imagine that we release a terraform plugin that needs to work with ngrok, and therefore will re-use the ngrok credential to provision the ngrok secrets. If terraform ends up using the file provisioner we'll end up with `terraform plan --config=...`  which is not an actual terraform command, because the ngrok file provisioner also appends this config flag to the command line of the executable. That is why the ngrok file provisioner should just remain specific to the ngrok tool.